modules/isActive.js

```diff
@@ -48,7 +42,7 @@ function routeIsActive(pathname, activeRoutes, activeParams, indexOnly) {
     return false
 
   if (indexOnly)
-    return activeRoutes.length > 1 && activeRoutes[activeRoutes.length - 2].indexRoute === route
+    return activeRoutes.length > 1 && activeRoutes[activeRoutes.length - 1] === route.indexRoute
```

Comment:
This is behaving inconsistently for me on a SSR.
Client correctly sets activeRoutes and renders fine.
Server doesn't agree... adding the history as one of the active routes.

server:
```es6
activeRoutes =
   [ { history: undefined, childRoutes: [Object] },
     { path: '/',
       component: [Function: App],
       indexRoute: [Object],
       childRoutes: [Object] },
     { component: [Function: Home] } ]
```

And so the previous code worked, but not this update.

https://github.com/rackt/react-router/issues/1810#issuecomment-147409066